src/expr/macro/src/lib.rs

```diff
@@ -24,6 +24,274 @@ mod parse;
 mod types;
 mod utils;
 
+/// Defining expression from a Rust Function.
+///
+/// The following example demonstrates a simple usage:
+///
+/// ```ignore
+/// #[function("add(int32, int32) -> int32")]
+/// fn add(x: i32, y: i32) -> i32 {
+///     x + y
+/// }
+/// ```
+///
+/// ## Function Signature
+///
+/// Each function must have a signature, specified in the `function("...")` part of the macro
+/// invocation. The signature follows this pattern:
+/// ```
+/// name([arg_types],*) -> [setof] return_type
+/// ```
+/// Where `name` is the function name, which must match the function name defined in `prost`.
+///
+/// The allowed data types are listed in the `name` column of the appendix's type matrix. Wildcards
+/// or `auto` can also be used, as explained below.
+///
+/// When `setof` appears before the return type, this indicates that the function is a set-returning
+/// function (table function), meaning it can return multiple values instead of just one. For more
+/// details, see the section on table functions.
+///
+/// ### Multiple Function Definitions
+///
+/// Multiple `#[function]` macros can be applied to a single generic Rust function to define
+/// multiple SQL functions of different types. For example:
+///
+/// ```ignore
+/// #[function("add(int16, int16) -> int16")]
+/// #[function("add(int32, int32) -> int32")]
+/// #[function("add(int64, int64) -> int64")]
+/// fn add<T: Add>(x: T, y: T) -> T {
+///     x + y
+/// }
+/// ```
+///
+/// ### Type Expansion
+///
+/// Types can be automatically expanded to multiple types using wildcards. Here are some examples:
+///
+/// - `*`: expands to all types.
+/// - `*int`: expands to int16, int32, int64.
+/// - `*float`: expands to float32, float64.
+///
+/// For instance, `#[function("count(*int) -> int64")]` will be expanded to the following three
+/// functions:
+///
+/// ```ignore
+/// #[function("count(int16) -> int64")]
+/// #[function("count(int32) -> int64")]
+/// #[function("count(int64) -> int64")]
```

Comment:
This is a fake example. It might be better to replace `count` with a real function 🤡